File: DeleteMiddleNode/DeleteMiddleNode.java
import java.util.*;
import java.io.*;

import crio.ds.List.ListNode;


class DeleteMiddleNode {
    public static ListNode get_kth_element(ListNode head, int k) {
        for(int i = 1 ; i < k ; ++i) {
            head = head.next;
        }
        return head;
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        ArrayList<Integer> list = new ArrayList<Integer>();
        for(int i = 0 ; i < n ; ++i) {
            int x = scanner.nextInt();
            list.add(x);
        }
        int k = scanner.nextInt();
        scanner.close();

        ListNode head = ListNode.createList(list);
        ListNode node = get_kth_element(head, k);
        Solution.deleteMiddleNode(node);

        List<Integer> res = ListNode.extractList(head);
        for(int element: res) {
            System.out.print(element + " ");
        }
        System.out.println();
    }
}

File: DeleteMiddleNode/DeleteMiddleNode.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"


ListNode* getKthNode(ListNode* head, int k) {
    if(k == 1) {
        return head;
    }
    return getKthNode(head->next , k - 1);
}

int main() {
    int n;
    cin >> n;
    vector<int> list(n);
    for(int i = 0 ; i < n ; ++i) {
        cin >> list[i];
    }
    int k;
    cin >> k;

    ListNode* head = ListNode().createList(list);
    ListNode* nodeToDelete = getKthNode(head , k);
    deleteMiddleNode(nodeToDelete);
    vector<int> resultList = ListNode().extractList(head);
    for(int i = 0 ; i < resultList.size() ; ++i) {
        cout << resultList[i] << " ";
    }
    cout << "\n";
    return 0;
}
